pkg/api/pod/util.go

```diff
@@ -385,6 +385,19 @@ func dropDisabledFields(
 		})
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) && !lifecycleTypeInUse(oldPodSpec) {
```

Comment:
I think you want to check this condition on each container and initContainer.  You don't want to allow new uses to this field on container B just because container A happened to use it.

It's really minor corner case, I admit

Reply:
How would you be sure that the container in the new pod spec is the same as the container in the old pod spec? Both the name of the container and its position could be changed in an update. 

e.g
old pod
```yaml
containers:
- name: usesSidecar
  image: sidecar
  lifecycle: sidecar
- name: normalContainer
  image: myapp
```
could change to
new pod
```yaml
containers:
- name: normalContainer
  image: myapp
- name: usesSidecarNewName
  image: sidecar
  lifecycle: sidecar
```
In this case I'm not sure how I would verify that container `usesSidecarNewName` used the sidecar lifecycle in the previous spec, and would therefore remove it, losing info and changing behaviour unintentionally. Maybe I'm missing something obvious.

Reply:
Good point.  The list declares "name" as the key, so the correct-est answer would be to let it remain on any container with the same name, but not allow it on containers with new names.

If that's easy, great.  It may simply not be worthwhile, in which case just comment to explain why this shortcut is taken - basically, "yes, it's not exactly right, but doing it right is not worth the effort".

Thoughts?

Reply:
ok thanks that was the clarification I needed, I have this got to work and added a test for it, so it will now check not if the Pod was using the lifecycle but if the container with the same name in the old pod spec was using it.